scripts/deploy.js

```diff
@@ -0,0 +1,30 @@
+// We require the Hardhat Runtime Environment explicitly here. This is optional
+// but useful for running the script in a standalone fashion through `node <script>`.
+//
+// When running the script with `npx hardhat run <script>` you'll find the Hardhat
+// Runtime Environment's members available in the global scope.
+import { ethers } from "hardhat";
+
+async function main() {
+  // Hardhat always runs the compile task when running scripts with its command
+  // line interface.
+  //
+  // If this script is run directly using `node` you may want to call compile
+  // manually to make sure everything is compiled
+  // await hre.run('compile');
+
+  // We get the contract to deploy
+  const Contract = await ethers.getContractFactory("ERC721A");
+  const contract = await Contract.deploy();
```

Comment:
Maybe I'm missing something, but don't upgradeable contracts use  `await upgrades.deployProxy(Contract)`? Reference: https://docs.openzeppelin.com/contracts/4.x/upgradeable#usage

Reply:
here's what I use, maybe it helps
```
      const instanceContract = await upgrades.deployProxy(CONTRACT, {
		kind: "uups",
		log: true,
		skipIfAlreadyDeployed: true,
	})
	
	with ugprading at
	
		// We get the contract to deploy
	const proxyAddress = "0x............................."
	const contract = await ethers.getContractFactory("CONTRACTV2")
	console.log("Upgrading contract...")
	const upgr = await upgrades.upgradeProxy(proxyAddress, contract)
	await upgr.deployed()
	console.log("Contract upgraded @ ", upgr.address)

	const contractImplementation = await getImplementationAddress(upgr.address)
	console.log("++++++++++++++++++++++++++++++++++++++++++++++++")
	console.log("Contract Implementation:", contractImplementation)
	console.log("++++++++++++++++++++++++++++++++++++++++++++++++")
	
	await sleep(20000)
	await hre.run("verify:verify", {
		address: contractImplementation,
		constructorArguments: [],
	})
```

Reply:
> here's what I use, maybe it helps
> 
> ```
> 
>       const instanceContract = await upgrades.deployProxy(CONTRACT, {
> 
> 		kind: "uups",
> 
> 		log: true,
> 
> 		skipIfAlreadyDeployed: true,
> 
> 	})
> 
> 	
> 
> 	with ugprading at
> 
> 	
> 
> 		// We get the contract to deploy
> 
> 	const proxyAddress = "0x............................."
> 
> 	const contract = await ethers.getContractFactory("CONTRACTV2")
> 
> 	console.log("Upgrading contract...")
> 
> 	const upgr = await upgrades.upgradeProxy(proxyAddress, contract)
> 
> 	await upgr.deployed()
> 
> 	console.log("Contract upgraded @ ", upgr.address)
> 
> 
> 
> 	const contractImplementation = await getImplementationAddress(upgr.address)
> 
> 	console.log("++++++++++++++++++++++++++++++++++++++++++++++++")
> 
> 	console.log("Contract Implementation:", contractImplementation)
> 
> 	console.log("++++++++++++++++++++++++++++++++++++++++++++++++")
> 
> 	
> 
> 	await sleep(20000)
> 
> 	await hre.run("verify:verify", {
> 
> 		address: contractImplementation,
> 
> 		constructorArguments: [],
> 
> 	})
> 
> ```

Big help, thanks friend

Reply:
Ya, that deploy script isn't for the upgradeable version. I forgot to include it. I'll add it, as well as the script to do the upgrade

Reply:
> here's what I use, maybe it helps
> 	const contractImplementation = await getImplementationAddress(upgr.address)
> ```

thanks, very useful. I had no idea you could do this. For anyway else it comes from here:
```
import { getImplementationAddress } from '@openzeppelin/upgrades-core';
```